sdk/webpubsub/azure-webpubsub-client/azure/webpubsub/client/_models.py

```diff
@@ -0,0 +1,567 @@
+        data: Any,
+        group: str,
+        from_user_id: str,
+        sequence_id: Optional[int] = None,
+    ) -> None:
+        self.kind: Literal["groupData"] = "groupData"
+        self.data_type = data_type
+        self.data = data
+        self.group = group
+        self.from_user_id = from_user_id
+        self.sequence_id = sequence_id
+
+
+class ServerDataMessage:
+    def __init__(self, data_type: WebPubSubDataType, data: Any, sequence_id: Optional[int] = None) -> None:
+        self.kind: Literal["serverData"] = "serverData"
+        self.data_type = data_type
+        self.data = data
+        self.sequence_id = sequence_id
+
+
+class SendToGroupMessage:
+    def __init__(
+        self,
+        data_type: WebPubSubDataType,
+        data: Any,
+        group: str,
+        no_echo: bool,
+        ack_id: Optional[int] = None,
+    ) -> None:
+        self.kind: Literal["sendToGroup"] = "sendToGroup"
+        self.data_type = data_type
+        self.data = data
+        self.group = group
+        self.no_echo = no_echo
+        self.ack_id = ack_id
+
+
+class OnRestoreGroupFailedArgs:
+    def __init__(self, group: str, error: Exception) -> None:
+        self.group = group
+        self.error = error
+
+
+WebPubSubMessage = TypeVar(
+    "WebPubSubMessage",
+    GroupDataMessage,
+    ServerDataMessage,
+    JoinGroupMessage,
+    LeaveGroupMessage,
+    ConnectedMessage,
+    DisconnectedMessage,
+    SendToGroupMessage,
+    SendEventMessage,
+    SequenceAckMessage,
+    AckMessage,
+)
+
+
+def get_pay_load(data: Any, data_type: WebPubSubDataType) -> Any:
+    if data_type == WebPubSubDataType.TEXT:
+        if not isinstance(data, str):
+            raise TypeError("Message must be a string.")
+        return data
+    if data_type == WebPubSubDataType.JSON:
+        return data
+    if data_type in (WebPubSubDataType.BINARY, WebPubSubDataType.PROTOBUF):
+        if isinstance(data, (bytes, bytearray)):
+            return base64.b64encode(data).decode()
+        raise TypeError("Message must be a bytes or bytearray")
+    raise TypeError(f"Unsupported dataType: {data_type}")
+
+
+def parse_payload(data: Any, data_type: WebPubSubDataType) -> Any:
+    if data_type == WebPubSubDataType.TEXT:
+        if not isinstance(data, str):
+            raise TypeError("Message must be a string when dataType is text")
+        return data
+    if data_type == "json":
+        return data
+    if data_type in ("binary", "protobuf"):
+        if isinstance(data, (bytes, bytearray)):
+            return data
+        return bytes(base64.b64decode(data))
+    raise TypeError(f"Unsupported dataType: {data_type}")
+
+
+class WebPubSubClientProtocol:
+    def __init__(self) -> None:
+        self.is_reliable_sub_protocol = False
+        self.name = ""
+
+    @staticmethod
+    def parse_messages(
+        raw_message: str,
+    ) -> Union[ConnectedMessage, DisconnectedMessage, GroupDataMessage, ServerDataMessage, AckMessage]:
```

Comment:
Will it be beneficial to define a base class for these xxMessage? so that the return type here can just be the base class

Reply:
I ever had a try like you said but it will cause mypy error.